acceptance/runAPIacceptance.sh

```diff
@@ -0,0 +1,211 @@
+    echo "       APPLICATION_PASSWORD, or specify it with the -p command line arg"
+    exit 1
+fi
+
+#
+# Get the current UID and GID. These are passed into the container for use in
+# creating a container-local user account so ownership of files created in the
+# container will match the user in the host OS.
+#
+CALLER_UID=`id -u`
+CALLER_GID=`id -g`
+
+if [ -n "${TAGS}" ]; then
+    for i in "${TAGS[@]}"
+    do
+        CUCUMBER_OPTS="${CUCUMBER_OPTS} --tags $i"
+    done
+fi
+
+if [ -n "${CUCUMBER_OPTS}" ]; then
+    echo "CUCUMBER_OPTS=${CUCUMBER_OPTS}"
+fi
+
+if [ "$interactive" == true ]; then
+    INTERACTIVE_OPTION="-i"
+    CMD="bash"
+elif [ `uname -s` == "Darwin" ]; then
+    # FIXME: This may work with a little testing and tweaking
+    echo "ERROR: not supported on Mac OS X"
+    exit 1
+else
+    CMD="runAPICucumber.sh ${CUCUMBER_OPTS}"
+fi
+
+parse_host() {
+    # Get the URL by removing the protocol
+    url="$(echo $1 | sed -e's,^\(.*://\)\(.*\),\2,g')"
+    # Get the host (in a simple-minded way)
+    host="$(echo $url | cut -d/ -f1)"
+    echo "$host"
+}
+
+# Lookup host IP from name. Use ping because we can't assume that nslookup is installed.
+lookup_host_ip() {
+    PING_RESULT=`ping -c 1 $1 2>&1 | grep ^PING `
+    if [ -z "${PING_RESULT}" ]; then
+        echo "ERROR: can't find target host IP address: 'ping -c 1 $1' failed" 1>&2
+        exit 1
+    fi
+
+    host_ip="$(echo ${PING_RESULT} | awk '{print $3}' | sed -r 's/\(//' | sed -r 's/\)//')"
+    echo "$host_ip"
+}
+
+TARGET_HOST=`parse_host ${APPLICATION_URL}`
+TARGET_HOST_IP=`lookup_host_ip ${TARGET_HOST}`
+
+HOST_IP=`hostname -i`
+if [[ $HOST_IP == 127* ]]; then
+    # serviced won't allow us to add a loopback address, so use docker's as an alternative
+    echo "Overriding default HOST_IP ($HOST_IP)"
+    HOST_IP=$(ip addr show docker0 | grep -w inet | awk {'print $2'} | cut -d/ -f1)
+fi
+echo "Using HOST_IP=$HOST_IP"
+
+#
+# If we're not running on Ubuntu 14.04 (the same version as the zenoss/capybara image), then
+# we need to mount the host's libdevmapper into the container since the container
+# doesn't have a compatible library
+#
+MOUNT_DEVMAPPER=false
+if [ -f /etc/redhat-release ]; then
+   MOUNT_DEVMAPPER=true
+else
+   UBUNTU_RELEASE=`lsb_release -r | cut -f2`
+   if [ "$UBUNTU_RELEASE" != "14.04" ]; then
+      MOUNT_DEVMAPPER=true
+   fi
+fi
+
+LIB_DEVMAPPER_MOUNT=""
+if [ "$MOUNT_DEVMAPPER" == "true" ]; then
+    LIBDEVMAPPER_SOURCE=`ldd ../serviced | grep libdevmapper | awk '{print $3}'`
+    LIBDEVMAPPER_TARGET=`echo $LIBDEVMAPPER_SOURCE | rev | cut -d '/' -f1 | rev`
+    LIB_DEVMAPPER_MOUNT="-v ${LIBDEVMAPPER_SOURCE}:/lib/x86_64-linux-gnu/${LIBDEVMAPPER_TARGET}"
+fi
+
+#
+# If not running in a jenkins job, provide default values for build number and job name
+#
+if [ -z "$BUILD_NUMBER" ]; then
+    BUILD_NUMBER="local-build"
+fi
+if [ -z "$JOB_NAME" ]; then
+    JOB_NAME="serviced-acceptance"
```

Comment:
To avoid potential collisions with UI acceptance, should probably be something like `serviced-api-acceptance`